### DISCUSSION TITLE
namedtuple rendered as list instead of dict

### DISCUSSION THREAD

**Discussion by Credentive-Sec** [encode/django-rest-framework#8796] [Potential Issue]
I have a django model that has a custom property which returns a list of named tuples generated with values_list(named=True). Example output:

[Row(over=UUID('1de7104d-0a6b-49fa-98e5-1538fcf9346d'), under=UUID('ef9e4095-a56e-43c5-a578-1559bd78fbba')), Row(over=UUID('1de7104d-0a6b-49fa-98e5-1538fcf9346d'), under=UUID('844dccc5-d15a-4c97-9c29-200facf59eda')),
<...>]

This is rendered in DRF as a list of ordinary tuples, eg:

            [
                "1de7104d-0a6b-49fa-98e5-1538fcf9346d",
                "ef9e4095-a56e-43c5-a578-1559bd78fbba"
            ],
            [
                "1de7104d-0a6b-49fa-98e5-1538fcf9346d",
                "844dccc5-d15a-4c97-9c29-200facf59eda"
            ],

Wheras I might expect it to be rendered as a dict, e.g.

            {
                "over" : "1de7104d-0a6b-49fa-98e5-1538fcf9346d",
                "under" : "ef9e4095-a56e-43c5-a578-1559bd78fbba"
            },
            {
                "over" : "1de7104d-0a6b-49fa-98e5-1538fcf9346d",
                 "under" : "844dccc5-d15a-4c97-9c29-200facf59eda"
            },

Is this intentional? Does it make more sense for a named tuple to be converted into a dict rather than a list?

**Comment by Credentive-Sec** [comment]
I have been able to create what I want by manually creating the dict:

    for sh in self.version_data.section_hierarchy.values_list("over", "under", named=True):
            section_hierearchy.append({"over": sh.over, "under": sh.under})

However, it still feels like DRF should render the named tuple as a dict automatically.

**Comment by auvipy** [comment]
so apparently there are some sort of serialization issue for UUID?

**Reply by Credentive-Sec** [reply]
Interesting, I hadn't thought of trying it with another data type - I used UUIDs in this particular model. I could try converting to string, maybe?

**Reply by Credentive-Sec** [reply]
Unless I am misunderstanding - are you suggesting that named tuples are converted to dicts unless the tuple contains UUIDs?
